compileopts/config.go

```diff
@@ -238,6 +238,9 @@ func (c *Config) BinaryFormat(ext string) string {
 		// More information:
 		// https://github.com/Microsoft/uf2
 		return "uf2"
+	case ".zip":
```

Comment:
Yeah, I am aware of it. But there are some assumption right into tinygo:
1) We can chose suffixes (we can't since `nrfutil` does only accept `zip` for dfu packages, sadly.
2) filename suffices correspond 1:1 to methods (as you have written `zip` is widely used for many things, so a `tinygo build -o xx.zip ` is not  unambigous.
3) filename suffices and methods have to have the same name  (coded in main.go)

I cannot work around 1 and 2 but will provide a change for (3).

Reply:
The filename suffix can remain `.zip`. That's fine. You can choose what format this zip file is using `BinaryFormat`, as suggested by @sago35.